Source/Microsoft.Teams.Apps.CompanyCommunicator.Send.Func/CompanyCommunicatorSendFunction.cs

```diff
@@ -330,161 +314,16 @@ public class CompanyCommunicatorSendFunction
                     statusCodeToStore = HttpStatusCode.InternalServerError;
                 }
 
-                await this.SaveSentNotificationData(
-                    messageContent.NotificationId,
-                    messageContent.UserDataEntity.AadId,
-                    totalNumberOfThrottles,
+                await this.manageResultDataService.ProccessResultDataAsync(
+                    notificationId: messageContent.NotificationId,
+                    aadId: messageContent.UserDataEntity.AadId,
+                    totalNumberOfThrottles: totalNumberOfThrottles,
                     isStatusCodeFromCreateConversation: false,
-                    statusCode: statusCodeToStore);
+                    statusCode: statusCodeToStore,
+                    errorMessage: e.GetType().ToString());
 
                 throw e;
```

Comment:
Just `throw;` to rethrow, else you overwrite the information from the original throw site.
(Please scrub for other instances where we rethrow an exception incorrectly.)